### PR TITLE
Bug fix: Fix DisableExecuteApiEndpoint bug fix

### PR DESCRIPTION
**Pull request by aaythapa**
### Issue #, if available
[2396](https://github.com/aws/serverless-application-model/issues/2396)

### Description of changes
The `DisableExecuteApiEndpoint` property in Serverless API didn't work if the `OpenApiVersion` was 3 because, when transformed, it didn't add the `x-amazon-apigateway-endpoint-configuration` extension to the Servers object as stated in  [this guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-endpoint-configuration.html). Now it adds the extension to the Servers object.

### Checklist

- [X] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [X] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [X] Update documentation
- [X] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by GavinZZ** [comment]
Out of curiosity, how did you generate the transform test input and output files? Did you try using the script `add_transform_test.py`?

**Comment by aaythapa** [comment]
> Out of curiosity, how did you generate the transform test input and output files? Did you try using the script `add_transform_test.py`?
Didn't use the script unfortunately

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -131,12 +132,25 @@ def add_disable_execute_api_endpoint_extension(self, disable_execute_api_endpoin
...
         :param boolean disable_execute_api_endpoint: Specifies whether clients can invoke your API by using the default execute-api endpoint.
         """
-        if not self._doc.get(self._X_ENDPOINT_CONFIG):
-            self._doc[self._X_ENDPOINT_CONFIG] = {}
-
         DISABLE_EXECUTE_API_ENDPOINT = "disableExecuteApiEndpoint"
         set_disable_api_endpoint = {DISABLE_EXECUTE_API_ENDPOINT: disable_execute_api_endpoint}
-        self._doc[self._X_ENDPOINT_CONFIG].update(set_disable_api_endpoint)
+
+        # Check if the OpenAPI version is 3.0
```
Comment (hoffa) [inline samtranslator/swagger/swagger.py:138]: A comment explaining why would be helpful.

File: samtranslator/swagger/swagger.py [hunk 2]
```diff
@@ -131,12 +132,25 @@ def add_disable_execute_api_endpoint_extension(self, disable_execute_api_endpoin
...
+        if self._doc.get("openapi") and SwaggerEditor.safe_compare_regex_with_string(  # type: ignore[no-untyped-call]
+            SwaggerEditor.get_openapi_version_3_regex(), self._doc["openapi"]  # type: ignore[no-untyped-call]
```
Comment (hoffa) [inline samtranslator/swagger/swagger.py:140]: Can you try typing the functions?

Comment (hoffa) [inline samtranslator/swagger/swagger.py:140]: Since this is copied from elsewhere, a static method abstracting it would be much clearer.

**Comment by GavinZZ** [comment]
> > Out of curiosity, how did you generate the transform test input and output files? Did you try using the script `add_transform_test.py`?
> 
> Didn't use the script unfortunately

That's alright. FYI, you can use the script for easier file generation. It will generate the 3 files each for one partition, add to the correct directories and add `REGIONAL` endpoint configuration for you.

**Review by GavinZZ — APPROVED** [review]
Other than the changes Chris mentioned, rest LGTM

File: samtranslator/model/apigateway.py [hunk 3]
```diff
@@ -132,13 +132,13 @@ def __init__(self, api_logical_id=None, response_parameters=None, response_templ
...
         # Defaults to Py27Dict() as these will go into swagger
-        self.response_parameters = response_parameters or Py27Dict()  # type: ignore[no-untyped-call]
-        self.response_templates = response_templates or Py27Dict()  # type: ignore[no-untyped-call]
+        self.response_parameters = response_parameters or Py27Dict()
+        self.response_templates = response_templates or Py27Dict()
         self.status_code = status_code_str
 
     def generate_swagger(self):  # type: ignore[no-untyped-def]
         # Applying Py27Dict here as this goes into swagger
-        swagger = Py27Dict()  # type: ignore[no-untyped-call]
```
Comment (hoffa) [inline samtranslator/model/apigateway.py:141]: Nice! Great to see all these removed, and good to see the strict typing has a ripple effect.

File: samtranslator/swagger/swagger.py (hunk 2 above)
Comment (aaythapa) [reply samtranslator/swagger/swagger.py]: added static method

**Review by hoffa — APPROVED** [review]

**Pull request merged**